src/main/kotlin/com/terraformation/backend/admin/AdminVotersController.kt

```diff
@@ -0,0 +1,251 @@
+
+    val projects = getAcceleratorProjects().associateBy { it.projectId }
+    try {
+      userIds?.forEach { voteStore.delete(projectId, projects[projectId]!!.phase, it) }
+      redirectAttributes.successMessage = "Voter removed."
+    } catch (e: Exception) {
+      log.error("Failed to remove voters from project $projectId", e)
+      redirectAttributes.failureMessage = "Failed to bulk remove voter: ${e.message}"
+    }
+
+    return redirectToVoters(email)
+  }
+
+  @PostMapping("/voters/default/add")
+  fun addDefaultVoter(
+      @RequestParam email: String?,
+      @RequestParam userId: UserId,
+      redirectAttributes: RedirectAttributes,
+  ): String {
+    requirePermissions { updateDefaultVoters() }
+
+    try {
+      defaultVoterStore.insert(userId)
+      redirectAttributes.successMessage = "Default voter added."
+    } catch (e: Exception) {
+      log.error("Failed to add default voter $userId", e)
+      redirectAttributes.failureMessage = "Failed to default add voter: ${e.message}"
+    }
+
+    return redirectToVoters(email)
+  }
+
+  @PostMapping("/voters/default/remove")
+  fun removeDefaultVoter(
+      @RequestParam email: String?,
+      @RequestParam userId: UserId,
+      redirectAttributes: RedirectAttributes,
+  ): String {
+    requirePermissions { updateDefaultVoters() }
+
+    try {
+      defaultVoterStore.delete(userId)
+      redirectAttributes.successMessage = "Default voter removed."
+    } catch (e: Exception) {
+      log.error("Failed to remove default voter $userId", e)
+      redirectAttributes.failureMessage = "Failed to remove default voter: ${e.message}"
+    }
+
+    return redirectToVoters(email)
+  }
+
+  @PostMapping("/voters/default/bulkRemove")
+  fun bulkRemoveVoter(
+      @RequestParam email: String?,
+      @RequestParam userIds: List<UserId>?,
+      redirectAttributes: RedirectAttributes,
+  ): String {
+    requirePermissions { updateDefaultVoters() }
+
+    try {
+      userIds?.forEach { defaultVoterStore.delete(it) }
+      redirectAttributes.successMessage = "Voter removed."
+    } catch (e: Exception) {
+      log.error("Failed to remove voters as default voters", e)
+      redirectAttributes.failureMessage = "Failed to bulk remove default voter: ${e.message}"
+    }
+
+    return redirectToVoters(email)
+  }
+
+  private fun getAcceleratorProjects(): List<AcceleratorProject> {
+    return dslContext
+        .select(
+            COHORTS.NAME,
+            COHORTS.PHASE_ID,
+            PROJECTS.ID,
+            PROJECTS.NAME,
+            PROJECT_VOTE_DECISIONS.VOTE_OPTION_ID)
+        .from(PROJECTS)
+        .join(PARTICIPANTS)
+        .on(PARTICIPANTS.ID.eq(PROJECTS.PARTICIPANT_ID))
+        .join(COHORTS)
+        .on(PARTICIPANTS.COHORT_ID.eq(COHORTS.ID))
+        .leftJoin(PROJECT_VOTE_DECISIONS)
+        .on(PROJECT_VOTE_DECISIONS.PROJECT_ID.eq(PROJECTS.ID))
+        .orderBy(COHORTS.ID, PROJECTS.ID)
+        .fetch { AcceleratorProject.of(it) }
+  }
```

Comment:
Not the worst thing ever, but I think it's generally good for database queries to live in store classes rather than in controllers just to keep the layering clean, make unit testing easier, and promote reuse.

Reply:
I agree. I am going to refactor this into a AcceleratorService and add in any permissions needed.